File: arrayz/easy/is_sorted.cpp
#include <iostream>
using namespace std;

bool isSorted(int arr[], int n) {
  for (int i = 1; i < n; i++) {
    if (arr[i] < arr[i - 1]) {
      return false;
    }
  }
  return true;
}

int main() {
  int arr[] = {1, 2, 3, 4, 5};
  int n = sizeof(arr) / sizeof(arr[0]);
  bool ans = isSorted(arr, n);
  if (ans) {
    cout << "SORTED" << endl;
  } else {
    cout << "NOT SORTED" << endl;
  }
}

File: arrayz/easy/is_sorted.rs
// Single traversal - O(N)
fn is_sorted(arr: &[i32]) -> bool {
    for i in 0..arr.len() - 1 {
        if arr[i] > arr[i + 1] {
            return false;
        }
    }
    true
}

fn main() {
    let arr: [i32; 5] = [1, 2, 3, 4, 5];
    println!("{}", is_sorted(&arr));
}
